### ISSUE TITLE
Luakit segfaults with free(): invalid pointer

### ISSUE THREAD

**Issue by carboniris** [aidanholm/luakit#183]
I'm running a freshly installed luakit from the AUR package, and when running luakit I get a segfault. Running with --verbose shows the error: `*** Error in luakit: free(): invalid pointer: 0x00007f73528e3ce1 ***`

**Comment by aidanholm** [comment]
What's the command line you used: `luakit --verbose`? Did it crash immediately on startup? If you can reproduce the crash, could you provide a backtrace with coredumpctl?

**Comment by carboniris** [comment]
It crashes immediately without showing a window. Running with --verbose gives the output:

```
[    0.186303] V: website_data_dir_init:83: base_data_directory:                 (null)
[    0.186422] V: website_data_dir_init:84: base_cache_directory:                (null)
[    0.186466] V: website_data_dir_init:85: disk_cache_directory:                /home/me/.cache/luakit
[    0.186507] V: website_data_dir_init:86: indexeddb_directory:                 /home/me/.local/share/luakit/indexeddb
[    0.186546] V: website_data_dir_init:87: local_storage_directory:             /home/me/.local/share/luakit/local_storage
[    0.186585] V: website_data_dir_init:88: offline_application_cache_directory: /home/me/.local/share/luakit/applications
[    0.186623] V: website_data_dir_init:89: websql_directory:                    /home/me/.local/share/luakit/websql
Stylesheet directory '/home/me/.local/share/luakit/styles/' doesn't exist, not loading user styles...
[    0.275220] V: luaH_widget_set_type:229: created widget of type: window
[    0.275650] V: luaH_widget_set_type:229: created widget of type: eventbox
[    0.276247] V: luaH_widget_set_type:229: created widget of type: vbox
[    0.277386] V: luaH_widget_set_type:229: created widget of type: vpaned
[    0.278366] V: luaH_widget_set_type:229: created widget of type: notebook
[    0.278666] V: luaH_widget_set_type:229: created widget of type: hbox
[    0.278881] V: luaH_widget_set_type:229: created widget of type: eventbox
[    0.279109] V: luaH_widget_set_type:229: created widget of type: hbox
[    0.279303] V: luaH_widget_set_type:229: created widget of type: eventbox
[    0.280629] V: luaH_widget_set_type:229: created widget of type: label
[    0.280942] V: luaH_widget_set_type:229: created widget of type: label
[    0.281228] V: luaH_widget_set_type:229: created widget of type: label
[    0.281436] V: luaH_widget_set_type:229: created widget of type: eventbox
[    0.281673] V: luaH_widget_set_type:229: created widget of type: hbox
[    0.281857] V: luaH_widget_set_type:229: created widget of type: eventbox
[    0.282115] V: luaH_widget_set_type:229: created widget of type: label
[    0.282369] V: luaH_widget_set_type:229: created widget of type: label
[    0.282616] V: luaH_widget_set_type:229: created widget of type: label
[    0.282888] V: luaH_widget_set_type:229: created widget of type: label
[    0.283152] V: luaH_widget_set_type:229: created widget of type: vbox
[    0.283579] V: luaH_widget_set_type:229: created widget of type: overlay
[    0.283818] V: luaH_widget_set_type:229: created widget of type: hbox
[    0.284012] V: luaH_widget_set_type:229: created widget of type: eventbox
[    0.284297] V: luaH_widget_set_type:229: created widget of type: label
[    0.303563] V: luaH_widget_set_type:229: created widget of type: entry
[    0.303924] V: luaH_widget_set_type:229: created widget of type: vbox
[    0.308737] V: luaH_widget_set_type:229: created widget of type: scrolled
[    0.309074] V: luaH_widget_set_type:229: created widget of type: hbox
[    0.325006] V: luaH_widget_set_type:229: created widget of type: label
[    0.325855] V: luaH_widget_set_type:229: created widget of type: label
[    0.329254] I: web_context_init_finish:113: Web process count: 0
[    0.385630] V: luaH_widget_set_type:229: created widget of type: webview
[    0.386545] V: luaH_widget_set_type:229: created widget of type: eventbox
[    0.386897] V: luaH_widget_set_type:229: created widget of type: label
Segmentation fault (core dumped)
```

And here's the output from `coredumpctl dump` afterwards:

```
           PID: 8269 (luakit)
           UID: 1000 (me)
           GID: 1001 (me)
        Signal: 11 (SEGV)
     Timestamp: Sat 2017-02-18 08:45:36 PST (3min 57s ago)
  Command Line: luakit --verbose
    Executable: /usr/bin/luakit
 Control Group: /user.slice/user-1000.slice/session-c1.scope
          Unit: session-c1.scope
         Slice: user-1000.slice
       Session: c1
     Owner UID: 1000 (me)
       Boot ID: 222a4933ec1d4942abff018dd0feee34
    Machine ID: f428abecfe204490bd78219977660671
      Hostname: myhost
       Storage: /var/lib/systemd/coredump/core.luakit.1000.222a4933ec1d4942abff018dd0feee34.8269.1487436336000000000000.lz4
       Message: Process 8269 (luakit) of user 1000 dumped core.
                
                Stack trace of thread 8269:
                #0  0x00007fa2f1c01f55 malloc_printerr (libc.so.6)
                #1  0x00007fa2f1c027de _int_free (libc.so.6)
                #2  0x000000000041fec1 n/a (luakit)
                #3  0x00007fa2ed0b83c6 n/a (libluajit-5.1.so.2)
                #4  0x00007fa2ed0fdba0 lua_pcall (libluajit-5.1.so.2)
                #5  0x00000000004178f3 luaH_object_emit_signal (luakit)
                #6  0x00000000004245aa n/a (luakit)
                #7  0x00007fa2ed641f75 g_closure_invoke (libgobject-2.0.so.0)
                #8  0x00007fa2ed653f82 n/a (libgobject-2.0.so.0)
                #9  0x00007fa2ed65cbcc g_signal_emit_valist (libgobject-2.0.so.0)
                #10 0x00007fa2ed65cfaf g_signal_emit (libgobject-2.0.so.0)
                #11 0x00007fa2eec02f1b n/a (libgtk-3.so.0)
                #12 0x000000000041a051 n/a (luakit)
                #13 0x00007fa2ed0b83c6 n/a (libluajit-5.1.so.2)
                #14 0x00007fa2ed0fdba0 lua_pcall (libluajit-5.1.so.2)
                #15 0x0000000000415f37 luaH_loadrc (luakit)
                #16 0x0000000000425c0b luaH_parserc (luakit)
                #17 0x0000000000410468 main (luakit)
                #18 0x00007fa2f1bab291 __libc_start_main (libc.so.6)
                #19 0x000000000041054a _start (luakit)
                
                Stack trace of thread 8271:
                #0  0x00007fa2f1f3610f pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                #1  0x00007fa2e620d4cc __gthread_cond_wait (libstdc++.so.6)
                #2  0x00007fa2ece8aebd _ZN3WTF10ParkingLot21parkConditionallyImplEPKvRKNS_12ScopedLambdaIFbvEEERKNS3_IFvvEEENSt6chrono10time_pointINSC_3_V212steady_clockENSC_8durationIlSt5ratioILl1ELl1000000000EEEEEE (libjavascriptcoregtk-4.0.so.18)
                #3  0x00007fa2f032a3cf n/a (libwebkit2gtk-4.0.so.37)
                #4  0x00007fa2f032b6b0 n/a (libwebkit2gtk-4.0.so.37)
                #5  0x00007fa2ece8e145 n/a (libjavascriptcoregtk-4.0.so.18)
                #6  0x00007fa2ecec0eba n/a (libjavascriptcoregtk-4.0.so.18)
                #7  0x00007fa2f1f30454 start_thread (libpthread.so.0)
                #8  0x00007fa2f1c737df __clone (libc.so.6)
                
                Stack trace of thread 8275:
                #0  0x00007fa2f1c6a48d poll (libc.so.6)
                #1  0x00007fa2ebedb786 n/a (libglib-2.0.so.0)
                #2  0x00007fa2ebedbb12 g_main_loop_run (libglib-2.0.so.0)
                #3  0x00007fa2ecec3d60 _ZN3WTF7RunLoop3runEv (libjavascriptcoregtk-4.0.so.18)
                #4  0x00007fa2ecec299e n/a (libjavascriptcoregtk-4.0.so.18)
                #5  0x00007fa2ece8e145 n/a (libjavascriptcoregtk-4.0.so.18)
                #6  0x00007fa2ecec0eba n/a (libjavascriptcoregtk-4.0.so.18)
                #7  0x00007fa2f1f30454 start_thread (libpthread.so.0)
                #8  0x00007fa2f1c737df __clone (libc.so.6)
                
                Stack trace of thread 8272:
                #0  0x00007fa2f1f393ad accept (libpthread.so.0)
                #1  0x0000000000425608 n/a (luakit)
                #2  0x00007fa2ebf030d5 n/a (libglib-2.0.so.0)
                #3  0x00007fa2f1f30454 start_thread (libpthread.so.0)
                #4  0x00007fa2f1c737df __clone (libc.so.6)
                
                Stack trace of thread 8273:
                #0  0x00007fa2f1c6a48d poll (libc.so.6)
                #1  0x00007fa2ebedb786 n/a (libglib-2.0.so.0)
                #2  0x00007fa2ebedb89c g_main_context_iteration (libglib-2.0.so.0)
                #3  0x00007fa2ebedb8e1 n/a (libglib-2.0.so.0)
                #4  0x00007fa2ebf030d5 n/a (libglib-2.0.so.0)
                #5  0x00007fa2f1f30454 start_thread (libpthread.so.0)
                #6  0x00007fa2f1c737df __clone (libc.so.6)
                
                Stack trace of thread 8274:
                #0  0x00007fa2f1c6a48d poll (libc.so.6)
                #1  0x00007fa2ebedb786 n/a (libglib-2.0.so.0)
                #2  0x00007fa2ebedbb12 g_main_loop_run (libglib-2.0.so.0)
                #3  0x00007fa2ed94f316 n/a (libgio-2.0.so.0)
                #4  0x00007fa2ebf030d5 n/a (libglib-2.0.so.0)
                #5  0x00007fa2f1f30454 start_thread (libpthread.so.0)
                #6  0x00007fa2f1c737df __clone (libc.so.6)
                
                Stack trace of thread 8276:
                #0  0x00007fa2f1c6a48d poll (libc.so.6)
                #1  0x00007fa2ebedb786 n/a (libglib-2.0.so.0)
                #2  0x00007fa2ebedbb12 g_main_loop_run (libglib-2.0.so.0)
                #3  0x00007fa2ecec3d60 _ZN3WTF7RunLoop3runEv (libjavascriptcoregtk-4.0.so.18)
                #4  0x00007fa2ecec299e n/a (libjavascriptcoregtk-4.0.so.18)
                #5  0x00007fa2ece8e145 n/a (libjavascriptcoregtk-4.0.so.18)
                #6  0x00007fa2ecec0eba n/a (libjavascriptcoregtk-4.0.so.18)
                #7  0x00007fa2f1f30454 start_thread (libpthread.so.0)
                #8  0x00007fa2f1c737df __clone (libc.so.6)
                
                Stack trace of thread 8277:
                #0  0x00007fa2f1c6a48d poll (libc.so.6)
                #1  0x00007fa2ebedb786 n/a (libglib-2.0.so.0)
                #2  0x00007fa2ebedbb12 g_main_loop_run (libglib-2.0.so.0)
                #3  0x00007fa2ecec3d60 _ZN3WTF7RunLoop3runEv (libjavascriptcoregtk-4.0.so.18)
                #4  0x00007fa2ecec299e n/a (libjavascriptcoregtk-4.0.so.18)
                #5  0x00007fa2ece8e145 n/a (libjavascriptcoregtk-4.0.so.18)
                #6  0x00007fa2ecec0eba n/a (libjavascriptcoregtk-4.0.so.18)
                #7  0x00007fa2f1f30454 start_thread (libpthread.so.0)
                #8  0x00007fa2f1c737df __clone (libc.so.6)
                
                Stack trace of thread 8280:
                #0  0x00007fa2f1f364b8 pthread_cond_timedwait@@GLIBC_2.3.2 (libpthread.so.0)
                #1  0x00007fa2ece8acbc _ZN3WTF10ParkingLot21parkConditionallyImplEPKvRKNS_12ScopedLambdaIFbvEEERKNS3_IFvvEEENSt6chrono10time_pointINSC_3_V212steady_clockENSC_8durationIlSt5ratioILl1ELl1000000000EEEEEE (libjavascriptcoregtk-4.0.so.18)
                #2  0x00007fa2ece7ca42 _ZN3WTF5sleepEd (libjavascriptcoregtk-4.0.so.18)
                #3  0x00007fa2efa6456d n/a (libwebkit2gtk-4.0.so.37)
                #4  0x00007fa2ece8e145 n/a (libjavascriptcoregtk-4.0.so.18)
                #5  0x00007fa2ecec0eba n/a (libjavascriptcoregtk-4.0.so.18)
                #6  0x00007fa2f1f30454 start_thread (libpthread.so.0)
                #7  0x00007fa2f1c737df __clone (libc.so.6)
                
                Stack trace of thread 8270:
                #0  0x00007fa2f1f3610f pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                #1  0x00007fa2e620d4cc __gthread_cond_wait (libstdc++.so.6)
                #2  0x00007fa2ececa6bc _ZN7bmalloc9AsyncTaskINS_4HeapEMS1_FvvEE13threadRunLoopEv (libjavascriptcoregtk-4.0.so.18)
                #3  0x00007fa2ececa809 _ZN7bmalloc9AsyncTaskINS_4HeapEMS1_FvvEE16threadEntryPointEPS4_ (libjavascriptcoregtk-4.0.so.18)
                #4  0x00007fa2e621358f execute_native_thread_routine (libstdc++.so.6)
                #5  0x00007fa2f1f30454 start_thread (libpthread.so.0)
                #6  0x00007fa2f1c737df __clone (libc.so.6)
Refusing to dump core to tty (use shell redirection or specify --output).
```

**Comment by aidanholm** [comment]
Fix pushed at 071e786, thanks for the bug report!
